cloudnet-modules/cloudnet-bridge/src/main/java/de/dytanic/cloudnet/ext/bridge/bukkit/listener/BukkitPlayerListener.java

```diff
@@ -46,7 +46,23 @@ public void handle(PlayerLoginEvent event) {
         String currentTaskName = Wrapper.getInstance().getServiceId().getTaskName();
         ServiceTask serviceTask = Wrapper.getInstance().getServiceTaskProvider().getServiceTask(currentTaskName);
 
-        if (serviceTask != null && serviceTask.isMaintenance() && !player.hasPermission("cloudnet.bridge.maintenance")) {
+        if (serviceTask == null) {
+            return;
+        }
+
+        // if the service has a field "requiredPermission" and the field is not null or empty and
+        // the player has the has not the permission of that field -> disconnect him
+        if (serviceTask.getProperties().contains("requiredPermission") &&
+                serviceTask.getProperties().get("requiredPermission", String.class) != null &&
+                !serviceTask.getProperties().get("requiredPermission", String.class).equals("null") &&
+                !serviceTask.getProperties().get("requiredPermission", String.class).replace(" ", "").equals("") &&
```

Comment:
Please create a variable for the required permission and only check for nullability, it's ok when the permission contains spaces or is actually "null".

Reply:
And just use getString instead of get(String.class)

Reply:
I did that, the complete "null" string is sadly not possible without a own object in the jsonDocument, or I didn't found them. Please correct me on this one